manta-js/package/sdk.ts

```diff
@@ -0,0 +1,626 @@
+const SIGNER_URL = config.SIGNER_URL;
+const PRIVATE_ASSET_PREFIX = "p";
+const NATIVE_TOKEN_ASSET_ID = "1";
+
+/// The Envrionment that the sdk is configured to run for, if development
+/// is selected then it will attempt to connect to a local node instance.
+/// If production is selected it will attempt to connect to actual node.
+export enum Environment {
+    Development = "DEV",
+    Production = "PROD"
+}
+
+/// Supported networks.
+export enum Network {
+    Dolphin = "Dolphin",
+    Calamari = "Calamari",
+    Manta = "Manta"
+}
+
+/// MantaPrivateWallet class
+export class MantaPrivateWallet implements IMantaPrivateWallet {
+
+    api: ApiPromise;
+    wasm: any;
+    wasmWallet: Wallet;
+    network: Network;
+    environment: Environment;
+    wasmApi: wasmApi;
+    walletIsBusy: boolean;
+
+    constructor(api: ApiPromise, wasm: any, wasmWallet: Wallet, network: Network, environment: Environment, wasmApi: wasmApi) {
+        this.api = api;
+        this.wasm = wasm;
+        this.wasmWallet = wasmWallet;
+        this.network = network;
+        this.environment = environment;
+        this.wasmApi = wasmApi;
+        this.walletIsBusy = false;
+    }
+
+    ///
+    /// MantaPrivateWallet Initialization
+    ///
+
+    /// Initializes the MantaPrivateWallet class, given an optional address, this will be used
+    /// for specifying which polkadot.js address to use upon initialization if there are several.
+    /// If no address is specified then the first polkadot.js address will be used.
+    static async init(env: Environment, network: Network): Promise<MantaPrivateWallet> {
+        const { api } = await MantaPrivateWallet._initApi(env, network);
+        const { wasm, wasmWallet, wasmApi } = await MantaPrivateWallet._initWasmSdk(api);
+        const sdk = new MantaPrivateWallet(api,wasm,wasmWallet,network,env,wasmApi);
+        return sdk;
+    }
+
+    /// Private helper method for internal use to initialize the Polkadot.js API with web3Extension.
+    static async _initApi(env: Environment, network: Network): Promise<InitApiResult> {
+        const provider = new WsProvider(envUrl(env, network));
+        const api = await ApiPromise.create({ provider, types, rpc });
+        const [chain, nodeName, nodeVersion] = await Promise.all([
+            api.rpc.system.chain(),
+            api.rpc.system.name(),
+            api.rpc.system.version()
+        ]);
+        
+        console.log(`You are connected to chain ${chain} using ${nodeName} v${nodeVersion}`);
+
+        return {
+            api
+        };
+    }
+
+    /// Private helper method for internal use to initialize the initialize manta-wasm-wallet.
+    static async _initWasmSdk(api: ApiPromise): Promise<InitWasmResult> {
+        const wasm = await import('manta-wasm-wallet');
+        const wasmSigner = new wasm.Signer(SIGNER_URL);
+        const wasmApiConfig = new ApiConfig(
+            api, DEFAULT_PULL_SIZE, DEFAULT_PULL_SIZE
+        );
```

Comment:
We will soon want a way to optionally specify a custom ApiConfig that contains an `errorCallback`, `pullCallback`, and different pull sizes. This can be optional parameters to `init`. This also means we will want to remove `api` from the config class, since this library always handles initializing the api itself, and it really doesn't belong in a configuration object anyway. So initializing `wasmApi` would look like `new Api(api, wasmApiConfig);`